test/puppetlabs/puppetdb/amq_migration_test.clj

```diff
@@ -52,6 +54,43 @@
     (mapv (comp str #(.toAbsolutePath %))
           (-> path-stream .iterator iterator-seq))))
 
+(defn processed-state []
+  (reduce (fn [acc metric]
+            (assoc acc
+                   (keyword metric)
+                   (->> ["global" metric]
+                        (meters/meter (get-in metrics-registries [:mq :registry]))
+                        meters/rates
+                        :total)))
+          {} ["processed" "retried" "discarded"]))
+
+(defn await-processed [await-how-many starting-processed-state timeout-ms]
+  (let [stop-time (+ (System/currentTimeMillis)
+                     timeout-ms)
+        {prev-processed :processed
+         prev-retried :retried
+         prev-discarded :discarded} starting-processed-state]
+    (loop []
+      (let [{:keys [processed retried discarded]} (processed-state)]
+        (cond
+
+          (> (System/currentTimeMillis) stop-time)
+          (throw (Exception. "Timed out waiting for message to be processed"))
+
+          (< prev-retried retried)
+          (throw (Exception. "Unexepcted retry of message"))
+
+          (< prev-discarded discarded)
+          (throw (Exception. "Unexpected message discard"))
+
+          (>= processed (+ await-how-many prev-processed))
+          true
+
+          :else
+          (do
+            (Thread/sleep 1000)
```

Comment:
Yeah, this was unintentional. I had it as 10 ms but in trying to debug this test on faster machines I bumped it up, I'll put it back to what I had before.